api/server/routes/__tests__/mcp.spec.js

```diff
@@ -1307,4 +1309,75 @@ describe('MCP Routes', () => {
       expect(response.headers.location).toContain('/oauth/success');
     });
   });
+
+  describe('Cache Isolation Integration Tests', () => {
+    it('should isolate MCP tool cache between different users for same server', async () => {
+      const serverName = 'test-server';
+
+      // Simulate User A caching tools
+      const userATools = {
+        tool_a_mcp_testserver: {
+          name: 'tool_a',
+          description: 'User A tool',
+        },
+      };
+
+      // Simulate User B caching different tools
+      const userBTools = {
+        tool_b_mcp_testserver: {
+          name: 'tool_b',
+          description: 'User B tool',
+        },
+      };
+
+      setCachedTools.mockResolvedValue(true);
+
+      // Mock getMCPServerTools to simulate proper cache isolation
+      getMCPServerTools.mockImplementation((userId, server) => {
+        if (userId === 'user-a' && server === serverName) {
+          return Promise.resolve(userATools);
+        }
+        if (userId === 'user-b' && server === serverName) {
+          return Promise.resolve(userBTools);
+        }
+        return Promise.resolve(null);
+      });
+
+      // User A's tools should be isolated from User B
+      const resultA = await getMCPServerTools('user-a', serverName);
+      const resultB = await getMCPServerTools('user-b', serverName);
+
+      expect(resultA).toEqual(userATools);
+      expect(resultB).toEqual(userBTools);
+      expect(resultA).not.toEqual(resultB);
+
+      expect(getMCPServerTools).toHaveBeenCalledWith('user-a', serverName);
+      expect(getMCPServerTools).toHaveBeenCalledWith('user-b', serverName);
```

Comment:
This is actually testing the mock of `getMCPServerTools`. I deleted the actual `getMCPServerTools` function and this test still passed.